autotest/ogr/ogr_mssqlspatial.py

```diff
@@ -600,6 +600,67 @@ def test_ogr_mssqlspatial_bulk_insert():
 #
 
 
+@pytest.mark.require_driver("GPKG")
+def test_binary_field_bcp():
+    """Test for issue GH #3040 MSSQLSpatial: Cannot write binary field with MSSQLSPATIAL_USE_BCP = TRUE"""
+
```

Comment:
to skip the test when the driver is compiled, but the connection string not defined: (ideally we'd have a module-level pytest.skip.if() to do that, like in other similar drivers that require specific parameters to be passed)

```suggestion

    if gdaltest.mssqlspatial_ds is None:
        pytest.skip()

```

And you'd need to solve conflict with master now that your other MSSQL PR has been merged